src/SignalR/clients/cpp/src/signalrclient/negotiate.cpp

```diff
@@ -0,0 +1,103 @@
+
+            client.send(negotiate_url, request, [tce](http_response http_response, std::exception_ptr exception)
+            {
+                if (exception != nullptr)
+                {
+                    tce.set_exception(exception);
+                    return;
+                }
+
+                if (http_response.status_code != 200)
+                {
+                    tce.set_exception(signalr_exception("negotiate failed with status code " + std::to_string(http_response.status_code)));
+                    return;
+                }
+
+                try
+                {
+                    auto negotiation_response_json = web::json::value::parse(utility::conversions::to_string_t(http_response.content));
+
+                    negotiation_response response;
+
+                    if (negotiation_response_json.has_field(_XPLATSTR("error")))
+                    {
+                        response.error = utility::conversions::to_utf8string(negotiation_response_json[_XPLATSTR("error")].as_string());
+                        tce.set(std::move(response));
+                        return;
+                    }
+
+                    if (negotiation_response_json.has_field(_XPLATSTR("connectionId")))
+                    {
+                        response.connectionId = utility::conversions::to_utf8string(negotiation_response_json[_XPLATSTR("connectionId")].as_string());
+                    }
+
+                    if (negotiation_response_json.has_field(_XPLATSTR("availableTransports")))
+                    {
+                        for (auto transportData : negotiation_response_json[_XPLATSTR("availableTransports")].as_array())
+                        {
+                            available_transport transport;
+                            transport.transport = utility::conversions::to_utf8string(transportData[_XPLATSTR("transport")].as_string());
+
+                            for (auto format : transportData[_XPLATSTR("transferFormats")].as_array())
+                            {
+                                transport.transfer_formats.push_back(utility::conversions::to_utf8string(format.as_string()));
+                            }
+
+                            response.availableTransports.push_back(transport);
+                        }
+                    }
+
+                    if (negotiation_response_json.has_field(_XPLATSTR("url")))
+                    {
+                        response.url = utility::conversions::to_utf8string(negotiation_response_json[_XPLATSTR("url")].as_string());
+
+                        if (negotiation_response_json.has_field(_XPLATSTR("accessToken")))
+                        {
+                            response.accessToken = utility::conversions::to_utf8string(negotiation_response_json[_XPLATSTR("accessToken")].as_string());
+                        }
+                    }
+
+                    if (negotiation_response_json.has_field(_XPLATSTR("ProtocolVersion")))
```

Comment:
Is this the typical check you do for SignalR vs SignalR Core.

Reply:
Yes